src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -148,6 +216,56 @@ private static bool Quit(string title, string text)
             return n == 0;
         }
 
+        private void FilterData(ApplicationData applicationData, string filter, int[] columnWidths, int offset, Label filterError)
+        {
+            var items = new List<string>();
+            filterError.Text = " ";
+            filterError.ColorScheme = Colors.Base;
+            filterError.Redraw(filterError.Bounds);
+
+            foreach (DataTableRow dataTableRow in applicationData.DataTable.Data)
+            {
+                bool match = false;
+                var valueList = new List<string>();
+                foreach (var dataTableColumn in applicationData.DataTable.DataColumns)
+                {
+                    string dataValue = dataTableRow.Values[dataTableColumn.ToString()].DisplayValue;
+
+                    if (!match)
+                    {
+                        try
+                        {
+                            if (Regex.IsMatch(dataValue, filter, RegexOptions.IgnoreCase))
```

Comment:
User input can be empty/null filter that we could consider as ".*".

Reply:
will add